arctic/arctic.py

```diff
@@ -202,6 +213,44 @@ def list_libraries(self):
                         libs.append(coll[:-1 * len(self.METADATA_COLL) - 1])
         return libs
 
+    @mongo_retry
+    def invalidate_cache(self):
+        """
+        Deletes the data in the collection used to cache the result
+        of list_libraries
+        :return:
+        """
+        self._conn.meta_db.cache.drop()
+
+    @mongo_retry
+    def _list_libraries_cached(self):
+        """
+        Returns
+        -------
+        List of Arctic library names from a cached collection (global per mongo cluster) in mongo.
+        Long term list_libraries should have a use_cached argument.
+        TODO: Maybe convert this to a decorator so this code can be used in a generic way.
+        """
+        mongo_db = self._conn.meta_db
+        col_name = 'cache'
+        if col_name not in mongo_db.collection_names():
+            mongo_db.create_collection(col_name).create_index("date", expireAfterSeconds=3600)
+
+        cache_col = mongo_db[col_name]
```

Comment:
How does this interact with authentication on the cluster?

Reply:
I discussed this with @rob256, and we are going to create a role similar to list_databases and grant readWrite to all users for this db. Till the permissions are granted it should just throw an exception and use the slow path (the old list_libraries).